libraries/AP_HAL/Device.h

```diff
@@ -36,7 +36,8 @@ class AP_HAL::Device {
         SPEED_LOW,
     };
 
-    typedef void PeriodicHandle;
+    FUNCTOR_TYPEDEF(PeriodicCb, bool);
+    typedef void* PeriodicHandle;
```

Comment:
To give HAL implementations to implement it differently.  Even on Linux several times I stopped thinking that maybe it would be better to just return an int, in a filedescriptor-like manner, which in the end would be the idx in the vector maintained by the manager.  In the end I didn't do this, but I do see it as a good approach.